packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -238,7 +239,15 @@ export class MirrorNodeClient {
         }
 
         this.logger.error(new Error(error.message), `${requestIdPrefix} [${method}] ${path} ${effectiveStatusCode} status`);
-        throw new MirrorNodeClientError(error.message, effectiveStatusCode);
+
+        const mirrorError = new MirrorNodeClientError(error, effectiveStatusCode);
+
+        // we only need contract revert errors here as it's not the same as not supported
+        if (mirrorError.isContractReverted() && !mirrorError.isNotSupported() && !mirrorError.isNotSupportedSystemContractOperaton()) {
```

Comment:
I thought maybe it would end up looking like this but this is okay for now.
Maybe revisit how their tied together later
```suggestion
        if (mirrorError.isContractReverted() && !mirrorError.isNotSupportedSystemContractOperaton()) {
```

Reply:
Yes, I think one will be removed after we support precompiles, right ?